src/Discord.Net.Core/Entities/Interactions/MessageComponents/ComponentBuilder.cs

```diff
@@ -1415,6 +1418,9 @@ public string Value
                     throw new ArgumentOutOfRangeException(nameof(value), $"Value must not be longer than {MaxLength ?? LargestMaxLength}.");
                 if (value?.Length < (MinLength ?? 0))
                     throw new ArgumentOutOfRangeException(nameof(value), $"Value must not be shorter than {MinLength}");
+                if (Style == TextInputStyle.Short && value?.Contains('\n') == true)
```

Comment:
This validation shouldn't be placed in the property setter, as it might be used before setting the style of the text input.
It should be added to the `Build()` method instead, this way we can be sure the style is set to the right value.